soepy/exogenous_processes/partner.py

```diff
@@ -0,0 +1,32 @@
+"""This module reads in information on probabilities regarding the exogenous
+process of marriage."""
+import pandas as pd
+import numpy as np
+
+
+def gen_prob_partner(model_spec):
+    """ Generates a vector with length `num_periods` which contains
+    the probability to get a partner in the corresponding period."""
+
+    # Read data frame with information on probability to get a partner
+    # in every period
+    if model_spec.partner_info_file_name == "test":
+        index_levels = [list(range(0, model_spec.num_periods)), [0, 1, 2]]
```

Comment:
This creates `[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9], [0, 1, 2]]` rather than `[(0,0), (0,1), (0,2), (1,0), ...]`.
But thanks a lot to pointing me to `itertools`. Might be useful in other cases.